### PR TITLE
Change release job logic to handle more files

### PR DESCRIPTION
**Pull request by dtzar**
**What type of PR is this?**

/kind feature
/kind cleanup

**What this PR does / why we need it**:

Makes it so there doesn't have to be two different sequential PRs to do a new release.

### CODE REVIEW

**Review by mboersma — COMMENTED** [review]
/lgtm

File: .github/workflows/release.yaml
Comment (nojnhuh) [inline .github/workflows/release.yaml:38]: How do we expect this to behave when multiple CHANGELOG files are updated at the same time?

Comment (nojnhuh) [reply .github/workflows/release.yaml:38]: Is it possible to modify the `changed-files` action to only look at CHANGELOG files?

File: .github/workflows/release.yaml
Comment (nojnhuh) [inline .github/workflows/release.yaml:9]: Is this something we expect to keep, or is this just to get over the hump of this one failed job?

Comment (mboersma) [reply .github/workflows/release.yaml:9]: It's specifically so we don't have to create a no-op PR to trigger the job this time, but seems like it could useful to retain as an escape hatch.

File: .github/workflows/release.yaml
Comment (mboersma) [reply .github/workflows/release.yaml:38]: We expect it to misbehave in that case I guess: it would only process the first CHANGELOG.md file found. @dtzar and I thought this was unlikely, but this could be replaced with a proper loop if prefererred.

File: .github/workflows/release.yaml
Comment (nojnhuh) [reply .github/workflows/release.yaml:9]: Will the changed-files action pick up the CHANGELOG file then for a manually-triggered run?

Comment (mboersma) [reply .github/workflows/release.yaml:9]: Erm...maybe? I'm not sure actually. 🤦🏻 
I guess in that case we can fall back to creating another PR that makes a trivial change to the CHANGELOG (and revert `workflow_dispatch`).

File: .github/workflows/release.yaml
Comment (mboersma) [reply .github/workflows/release.yaml:38]: > Is it possible to modify the changed-files action to only look at CHANGELOG files?

Yes, according to the README https://github.com/tj-actions/changed-files?tab=readme-ov-file#usage-
```yaml
      - name: Get all changed markdown files
        id: changed-markdown-files
        uses: tj-actions/changed-files@v42
        with:
          # Avoid using single or double quotes for multiline patterns
          files: |
             **.md
```

File: .github/workflows/release.yaml
Comment (dtzar) [reply .github/workflows/release.yaml:9]: Good catch!  I just tested on my fork and you're right it doesn't detect files.  I will remove this.

**Review by nojnhuh — COMMENTED** [review]
/lgtm

**Comment by dtzar** [comment]
Hold off on this - test on my fork failed to detect any changes with this pattern

**Comment by dtzar** [comment]
Validated [this does work on my fork now](https://github.com/dtzar/cluster-api-provider-azure/actions/runs/8211304767/job/22460059373).

**Comment by dtzar** [comment]
Appreciate squash-merge so I don't have to squash manually 🙏

File: .github/workflows/release.yaml
Comment (nojnhuh) [inline .github/workflows/release.yaml:39]: After we get this PR merged, it looks like the changed-files action has a built-in way to feed the list of files into a matrix that can spawn one job per file which might be able to make cutting multiple patches pretty slick.

https://github.com/tj-actions/changed-files/blob/257d47dfba22be3e0a17f6bad47ff07f7e76747c/.github/workflows/matrix-example.yml

File: .github/workflows/release.yaml
Comment (nojnhuh) [inline .github/workflows/release.yaml:38]: Out of curiosity, how did this fix the issue you were seeing if this was already matching the id of the step? Was `changed-files` here being interpreted as something else before?

**Review by nojnhuh — COMMENTED** [review]
/lgtm

File: .github/workflows/release.yaml
Comment (dtzar) [reply .github/workflows/release.yaml:38]: It perhaps was with the logic used when it wasn't the `-markdown` flavor.  I saw this in the examples.  It's possible removing the `| ` and using the same `CHANGELOG/**.md` logic without markdown would work, but this does work and seems like a better fit anyways since indeed we only care about markdown files.

**Comment by dtzar** [comment]
There was a merge conflict with the newer version of the action, so had to push another commit.

**Review by mboersma — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
